nomad/container.py

```diff
@@ -165,18 +167,28 @@ def is_stopped(self):
         return self._container.status_code == constants.CONTAINER_STOPPED
 
     @property
+    def lxd_name(self):
+        """ Returns the name of the container that is used in the scope of LXD.
+
+        This name id supposed to be unique among all the containers managed by LXD.
+        """
+        # Note: all container names must be a valid hostname! That is: maximum 63 characters, no
+        # dots, no digit at first position, made entirely of letters/digits/hyphens, ...
+        if not hasattr(self, '_lxd_name'):
+            lxd_name_prefix = '{project_name}-{name}'.format(
```

Comment:
oh, yes, I was forgetting about referencing containers in the CLI... You are right. If we make project name mandatory, we might as well do the same for containers. That would leave us with only (3) and (4) to do.